### PR TITLE
feat: asset player render from chain

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description

[MASK-548](https://mask.atlassian.net/browse/MASK-548)
[MASK-563](https://mask.atlassian.net/browse/MASK-563)
## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/NftImage.tsx [hunk 1]
```diff
@@ -1,55 +1,59 @@
...
+    contractAddress: string
+    tokenId: string
     fallbackImage: URL
 }
 export function NftImage(props: NftImageProps) {
-    const { token, fallbackImage } = props
+    const { chainId, contractAddress, tokenId, fallbackImage } = props
     const classes = useStylesExtends(useStyles(), props)
-
+    console.log({ props })
```
Comment (septs) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/NftImage.tsx]: plz remove `console.*`

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/NftList.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { List, ListItem, ListProps, Skeleton, Typography } from '@mui/material'
 import classnames from 'classnames'
 import type { FC, HTMLProps } from 'react'
 import { useI18N } from '../../../utils'
-import { NftImage } from './NftImage'
+// import { NftImage } from './NftImage'
```
Comment (septs) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/NftList.tsx]: plz clean unused code

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketNft.tsx [hunk 3]
```diff
@@ -335,7 +311,7 @@ export function RedPacketNft({ payload }: RedPacketNftProps) {
     }, [shareLink])
     //#endregion
 
-    if (isFailedToLoading)
+    if (Boolean(availabilityError))
```
Comment (septs) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketNft.tsx]:
```suggestion
    if (availabilityError)
```

File: packages/shared/src/constants.tsx [hunk 4]
```diff
@@ -8,4 +8,4 @@ export const SOCIAL_MEDIA_ICON_MAPPING: Record<string, ReactNode> = {
     'instagram.com': <InstagramColoredIcon />,
 }
 
-export const mediaViewerUrl = 'https://dimensiondev.github.io/Media-Viewer/index.html'
+export const mediaViewerUrl = 'https://192.168.3.101:8081/index.html'
```
Comment (septs) [inline packages/shared/src/constants.tsx]: ????

Comment (hanszhoou) [reply packages/shared/src/constants.tsx]: still under developing...

File: packages/shared/src/UI/components/AssetPlayer/index.tsx [hunk 5]
```diff
@@ -69,28 +64,34 @@ export const AssetPlayer = memo<AssetPlayerProps>(({ url, type, options, iconPro
     //#region setup iframe when url and options be changed
     const setIframe = useCallback(() => {
         // if iframe isn't be init or the load error has been existed
-        if (!ref.current || playerState === AssetPlayerState.ERROR) return
-        if (!url) {
+        if (!ref.current || playerState === AssetPlayerState.ERROR || playerState === AssetPlayerState.NORMAL) return
```
Comment (hanszhoou) [inline packages/shared/src/UI/components/AssetPlayer/index.tsx:76]: Reduce sending repeat requests, which causes the image flashes multiple times. 

cc @nuanyang233

File: packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx [hunk 6]
```diff
@@ -16,11 +16,29 @@ const useStyles = makeStyles()({
...
     },
-})
+    loadingIcon: {
+        width: 36,
+        height: 52,
+    },
+    errorIcon: {
+        width: 36,
+        height: 36,
+    },
```
Comment (nuanyang233) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx:40]: These are the same as the default styles of `AssetPlayer` and do not need to be defined repeatedly

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/NFTLuckyDropStyledAssetPlayer.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        width: 120,
+    },
+    loadingFailImage: {
+        height: 160,
+        width: 120,
+    },
+    loadingIcon: {
+        width: 36,
+        height: 52,
+    },
```
Comment (nuanyang233) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/NFTLuckyDropStyledAssetPlayer.tsx]: As above

File: packages/shared/src/UI/components/AssetPlayer/index.tsx [hunk 8]
```diff
@@ -17,32 +27,12 @@ interface AssetPlayerProps
...
-        display: 'flex',
-        justifyContent: 'center',
-        alignItems: 'center',
-        width: '100%',
-    },
-    loadingIcon: {
-        width: 36,
-        height: 52,
-    },
-}))
```
Comment (nuanyang233) [inline packages/shared/src/UI/components/AssetPlayer/index.tsx]: Why remove these？

File: packages/shared/src/UI/components/AssetPlayer/index.tsx (hunk 5 above)
Comment (nuanyang233) [reply packages/shared/src/UI/components/AssetPlayer/index.tsx:76]: This does not allow for resending the message after the prop is changed to make the iframe render the new asset. But it does not affect the current.

File: packages/shared/src/constants.tsx (hunk 4 above)
Comment (nuanyang233) [reply packages/shared/src/constants.tsx]: I recommend using proxy software when you are developing locally. such as [charles](https://www.charlesproxy.com/)

File: packages/shared/src/UI/components/AssetPlayer/index.tsx (hunk 8 above)
Comment (hanszhoou) [reply packages/shared/src/UI/components/AssetPlayer/index.tsx]: image


There're some NFT components that have very different UI.

File: packages/web3-shared/evm/hooks/useERC721TokenDetailedOwnerList.ts [hunk 9]
```diff
@@ -39,21 +39,6 @@ export function useERC721TokenDetailedOwnerList(contractDetailed: ERC721Contract
...
-            const info = await getERC721TokenAssetFromChain(nft.info.tokenURI)
-            if (info) nft.info = { ...info, ...nft.info, hasTokenDetailed: true, name: info.name ?? nft.info.name }
-            return nft
-        })
-
-        allListRef.current = (await Promise.allSettled(allRequest)).map((x, i) => {
-            if (x.status === 'fulfilled') return x.value
-            allListRef.current[i].info.hasTokenDetailed = true
-            return allListRef.current[i]
-        })
```
Comment (hanszhoou) [inline packages/web3-shared/evm/hooks/useERC721TokenDetailedOwnerList.ts:56]: Moved this fetch logic to [Media-Viewer](https://github.com/DimensionDev/Media-Viewer)

**Review by septs — COMMENTED** [review]
use svg version replace `nft_token_fallback.png`?

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx [hunk 10]
```diff
@@ -81,10 +81,10 @@ const useStyles = makeStyles()((theme) => {
             position: 'relative',
             display: 'flex',
             flexDirection: 'column',
-            borderRadius: 6,
+            borderRadius: 8,
             padding: 0,
             marginBottom: theme.spacing(2.5),
-            background: theme.palette.background.default,
+            background: theme.palette.mode === 'light' ? '#fff' : '#2F3336',
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx]: If it's possible, avoid dead-code color.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketNft.tsx (hunk 3 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketNft.tsx]: Vote +1 for this.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedpacketNftConfirmDialog.tsx [hunk 11]
```diff
@@ -287,21 +287,9 @@ export function RedpacketNftConfirmDialog(props: RedpacketNftConfirmDialogProps)
...
-                                        fallbackImage={new URL('./assets/nft_token_fallback.png', import.meta.url)}
-                                    />
-
-                                    <div className={classes.nftNameWrapper}>
-                                        <Typography className={classes.nftName} color="textSecondary">
-                                            {value.info.name}
-                                        </Typography>
-                                    </div>
-                                </ListItem>
+                                <div key={i.toString()}>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedpacketNftConfirmDialog.tsx]:
```suggestion
                                <div key={i}>
```

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx [hunk 12]
```diff
@@ -632,43 +634,15 @@ export function SelectNftTokenDialog(props: SelectNftTokenDialogProps) {
...
-                                                    className={classNames(
-                                                        classes.selectWrapper,
-                                                        tokenIdFilterList.length > 0 &&
-                                                            !tokenIdFilterList.includes(token.tokenId)
-                                                            ? classes.hide
-                                                            : '',
-                                                    )}
-                                                    key={i}>
-                                                    <NftImage
+                                                <div key={i.toString()}>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx]:
```suggestion
                                                <div key={i}>
```

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx [hunk 13]
```diff
@@ -311,25 +312,9 @@ export function RedPacketERC721Form(props: RedPacketERC721FormProps) {
...
-                                            {value.info.name}
-                                        </Typography>
-                                    </div>
-                                    <div className={classes.closeIconWrapperBack} onClick={() => removeToken(value)}>
-                                        <div className={classes.closeIconWrapper}>
-                                            <CloseIcon className={classes.closeIcon} />
-                                        </div>
-                                    </div>
-                                </ListItem>
+                                <div key={i.toString()}>
```
Comment (septs) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketERC721Form.tsx]:
```suggestion
                                <div key={i}>
```

**Review by guanbinrui — DISMISSED** [review]

**Comment by wenluomask** [comment]
Checked.

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/dashboard/src/pages/Wallets/components/CollectibleList/index.tsx [hunk 14]
```diff
@@ -151,11 +151,12 @@ export const CollectibleListUI = memo<CollectibleListUIProps>(
                 ) : (
                     <Box>
                         <div className={classes.root}>
-                            {dataSource.map((x) => (
+                            {dataSource.map((x, i) => (
                                 <div className={classes.card} key={x.id}>
```
Comment (septs) [inline packages/dashboard/src/pages/Wallets/components/CollectibleList/index.tsx]:
```suggestion
                                <div className={classes.card} key={i}>
```

**Review by Lanttcat — DISMISSED** [review]

**Review by Lanttcat — APPROVED** [review]

File: packages/shared/src/UI/components/AssetPlayer/index.tsx [hunk 15]
```diff
@@ -51,53 +49,107 @@ enum AssetPlayerState {
...
+                setHidden(true)
+                setPlayerState(AssetPlayerState.LOADING)
+                setTimeout(() => setHidden(false), 1000)
+            }
+            if ((message as ERC721TokenNameMsg).message?.type === 'name') {
+                props.setERC721TokenName?.((message as ERC721TokenNameMsg).message.name)
+            }
+            if ((message as ERC721TokenNameMsg).message?.type === 'sourceType') {
+                props.setSourceType?.((message as ERC721TokenNameMsg).message.name)
+            }
```
Comment (septs) [inline packages/shared/src/UI/components/AssetPlayer/index.tsx:114]: @Jack-Works approve?

Comment (septs) [inline packages/shared/src/UI/components/AssetPlayer/index.tsx:114]: @Jack-Works approve?

**Review by yanzhihong23 — APPROVED** [review]

File: packages/dashboard/src/pages/Wallets/components/CollectibleList/index.tsx [hunk 16]
```diff
@@ -151,11 +151,12 @@ export const CollectibleListUI = memo<CollectibleListUIProps>(
                 ) : (
                     <Box>
                         <div className={classes.root}>
-                            {dataSource.map((x) => (
-                                <div className={classes.card} key={x.id}>
+                            {dataSource.map((x, index) => (
+                                <div className={classes.card} key={index}>
```
Comment (UncleBill) [inline packages/dashboard/src/pages/Wallets/components/CollectibleList/index.tsx:155]: index as key is anti-pattern, prefer uniq id.

File: packages/shared/src/UI/components/AssetPlayer/index.tsx (hunk 15 above)
Comment (UncleBill) [reply packages/shared/src/UI/components/AssetPlayer/index.tsx:114]: We can omit `as` operator with https://www.typescriptlang.org/docs/handbook/release-notes/overview.html#type-guards-inferred-from-in-operator

**Pull request merged**